### PR TITLE
Feature 13067 case reference number on case and contact

### LINKED ISSUE #13067: Extend the contacts with Case Reference Number & add case reference number to cases
### Feature Description

To be able to add a case-reference number to risk contacts and cases we have to add it 
- in the database
- in the Create & Update Forms
- Display it in the grid
- Add it to the search, like we did for the [Search by NationalID is not consistently working in all locations](https://github.com/SORMAS-Foundation/SORMAS-Project/issues/13029) 

### Added Value/Benefit

With the case reference people who self-declare themselves as Risk Contacts can link themselves to a case without knowing more about the case or having to know more about the case.

### Acceptance Criteria

- [x] Case Reference is available in all relevant create and update forms
- [x] Case Reference is shown in the grids
- [x] Case Reference is searchable
- [x] Case Reference is included in the exports and imports
- [x] Case Reference is part of the data dictionary and data protection dictionary
- [ ] Case Reference is only shown when self reporting feature is enabled

### Implementation Details

_No response_

### Mockups

_No response_

### Additional Information

_No response_

### PR DESCRIPTION
**Pull request by leventegal-she**
Fixes #13067

### CODE REVIEW

**Comment by sormas-vitagroup** [comment]
SonarCloud analysis: https://sonarcloud.io/dashboard?id=SORMAS-Project&pullRequest=13088

File: sormas-api/src/main/java/de/symeda/sormas/api/caze/CaseExportDto.java [hunk 1]
```diff
@@ -643,6 +645,16 @@ public String getInternalToken() {
 	}
 
 	@Order(7)
+	@ExportTarget(caseExportTypes = {
+		CaseExportType.CASE_SURVEILLANCE,
+		CaseExportType.CASE_MANAGEMENT })
+	@ExportProperty(CaseDataDto.CASE_REFERENCE_NUMBER)
+	@ExportGroup(ExportGroupType.CORE)
+	public String getCaseReferenceNumber() {
+		return caseReferenceNumber;
```
Comment (carina29) [inline sormas-api/src/main/java/de/symeda/sormas/api/caze/CaseExportDto.java:654]: I think that caseReference should not be displayed in the export files (based on the acceptance criteria "Case Reference is only shown when self reporting feature is enabled") when FeatureType.SELF_REPORTING is disabled.

Currently, when  FeatureType.SELF_REPORTING is disabled, the column will not appear in the grid as it was requested ( Contacts and Cases) but will appear in the Detailed exports for Cases for Contacts.

image
image

image

**Review by carina29 — CHANGES_REQUESTED** [review]

**Comment by carina29** [comment]
There is a failing test: ArchitectureTest.dontUseFacadeProviderRule

**Review by carina29 — CHANGES_REQUESTED** [review]
Failing test

**Comment by sormas-vitagroup** [comment]
SonarCloud analysis: https://sonarcloud.io/dashboard?id=SORMAS-Project&pullRequest=13088

**Review by carina29 — APPROVED** [review]

**Pull request merged**